add flag validation for --account-id (#605)

anubhav06 opened the pull request:
Signed-off-by: Anubhav Gupta <mail.anubhav06@gmail.com>

## Describe your changes
Fixes a Bad Request error when running a Kubescape scan with an Account ID that is not a valid UUID

This PR is in continuation to #577 
I've added the flag validation to all the commands (and their respective subcommands as well) which use the `--account` flag.

The commands which are covered are as follows:
1. `scan`
-> Subcommands: `framework`, `control`
2. `download`
3.  `delete`
-> Subcommands: `exceptions`
4. `list`
5. `submit`
-> Subcommands: `results`, `rbac`, `exceptions`


## Screenshots - If Any (Optional)
**Some** examples: 

![Screenshot (219)](https://user-images.githubusercontent.com/43821031/189214973-b1fdc310-61a9-453c-8303-fd0414948e41.png)

![3](https://user-images.githubusercontent.com/43821031/189215000-160e7ac0-a67b-4b17-9989-4b1193d90cde.png)

![4](https://user-images.githubusercontent.com/43821031/189215003-6e210e5f-cdfd-405b-8c1e-9be41d79e2c2.png)


## This PR fixes:

* Resolved #605 

## Checklist before requesting a review
<!-- put an [x] in the box to get it checked -->

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

**Please open the PR against the `dev` branch (Unless the PR contains only documentation changes)**

dwertent reviewed (approved):
Well done :)

vladklokun commented:
Yup, well done!

While this solution works, I was wondering if we wanted to reconsider our approach. I’m iffy on the idea of having a separate validation function for each command down the line, since we have a lot of these commands, because this leads to duplicate code. And copy-pasting is a code smell.

I was thinking that we should reconsider how we build inputs to our commands and add validation methods to the structs themselves, instead of writing custom commands for each command. This way clients of `{scan,delete,list}Info` structs can just call their respective `Validate()` method and propagate the error.

Ideally, this would mean that `{scan,delete,list}Info` structs embed some `baseInfo` struct which has a base `Validate` method, and embedders can then override and extend it.

@dwertent Is this refactoring of validation something we want to pursue? Or should we just wait until multiple validation functions start limiting us?

dwertent commented:
@vladklokun  I think each command should have a separate validation function.
That being said, when we have repeating validations we should call a function for that (and not implement it every time).

@anubhav06 Please implement a function that will receive `account`, `clientID`, and `secretKey` and will validate the three values (if not empty) are in UUID format.

This function should be called from the commands validation function.

anubhav06 commented:
Hello @dwertent. 
1. Can you pls tell me where can I find my `clientID` and `secretKey` ? There seems to be no documentation for finding these.  


> This function should be called from the commands validation function

2. Do you mean the **flag** validation functions which I have added ? 
Currently they just receive the `accountID` and validates it. I can modify it to receive and validate `clientID` and `secretKey` as well.

Is that what you meant ?

anubhav06 commented:
- Added validation for `client-id` and `secret-key` as well 
- Added a separate function - `ValidateCredentials` , which can be called from any of the flag validation functions
- Modified the flag validation functions, to call the `ValidateCredentials` function,  for validating the credentials

Is this what you were asking for @dwertent ?

![Screenshot (223)](https://user-images.githubusercontent.com/43821031/190923379-7b3468d0-8b29-4d4d-8ca5-7791232de6e7.png)  

![Screenshot (224)](https://user-images.githubusercontent.com/43821031/190923384-af19dfe4-934e-4d37-9da4-da55696be691.png)

dwertent commented:
@anubhav06 Please take a look at my changes :)

anubhav06 commented:
> @anubhav06 Please take a look at my changes :)

Oh understood 😄
